### ISSUE TITLE
Int values are not always recognized.

### ISSUE THREAD

**Issue by lgarron** [catilac/plistwatch#3]
Repro on macOS 15.1.1:

1. Run: `plistwatch | grep -v "ContextStoreAgent"`
2. Open System Settings → Control Center → Menu Bar Only → Clock → Clock Options…
3. Change the "Show date" dropdown value.
4. Click "Done" and wait for the change to be written.

This writes a number to the `plist`, as can be verified by keeping `~/Library/Preferences/com.apple.menuextra.clock.plist` open in Xcode.

However, `plistwatch` shows e.g.:

```
defaults write "com.apple.menuextra.clock" "ShowDate" '2'
```

Running this command will clobber the int value with a string and cause it to be effectively ignored by the OS (in this case).

Although I want to note that `plistwatch` has saved me a lot more time than it has cost me, this particular issue took me a few weeks to really understand and debug before I could fix it for good.

**Comment by pjg11** [comment]
While the tool itself doesn't do this yet, you can add the type next to the number like so:

```
defaults write "com.apple.menuextra.clock" "ShowDate" -int '2'
```

This will get the number to be correctly recognized as an integer instead of a string.

I came across this issue while using the tool recently, so I tried implementing it ([fork](https://github.com/pjg11/plistwatch)). It supports `int`, `float`, `array` and `string` for now.

**Comment by lgarron** [comment]
> While the tool itself doesn't do this yet, you can add the type next to the number like so:

Yeah, fortunately that's not too hard if you know that something is not meant to be written as a string.

What *is* an issue, though, is if the command implies that it can be run as-is for non-strings.

I just checked out your fork, and it seems to work well for me! I'll probably be using that for now, thanks! 🤩

**Comment by lgarron** [comment]
> I came across this issue while using the tool recently, so I tried implementing it ([fork](https://github.com/pjg11/plistwatch)). It supports `int`, `float`, `array` and `string` for now.

On a second look, it looks like this is trying to establish the type by looking at the serialized value, which I don't think produces correct and secure results. I think using `defaults read-type` would be more appropriate.

**Comment by pjg11** [comment]
Thanks for letting me know about `read-type`! I [updated my fork](https://github.com/pjg11/plistwatch/tree/value-types) to use it to get the type instead.

**Comment by catilac** [comment]
@pjg11 hi piya! thank you for fixing that, I would love to pull in your changes into this branch. I'm so sorry I've had to neglect this code base for so long
